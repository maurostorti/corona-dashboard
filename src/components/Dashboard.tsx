import { Divider, Stack } from '@mui/material'
import '../App.css'

type DashboardProps = {
  confirmed: string
  recovered: string
  deaths: string
}
export default function Dashboard({ confirmed, recovered, deaths }: DashboardProps) {
  return (
    <Stack direction='row' divider={<Divider orientation='vertical' flexItem />} spacing={2}>
      <Stack>
        <h2 className='Data-title'> Confimed Cases</h2>
        <p className='Data-content'> {confirmed} </p>
      </Stack>
      <Stack>
        <h2 className='Data-title'> Recovered Cases</h2>
        <p className='Data-content'>{recovered}</p>
      </Stack>
      <Stack>
        <h2 className='Data-title'> Deaths</h2>
        <p className='Data-content'>{deaths}</p>
      </Stack>
    </Stack>
  )
}
