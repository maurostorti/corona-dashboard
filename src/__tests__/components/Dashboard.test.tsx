import { render, screen } from '@testing-library/react'
import Dashboard from '../../components/Dashboard'

test('it renders correctly confirmed', () => {
  const { asFragment } = render(<Dashboard confirmed='123' recovered='456' deaths='789' />)
  expect(asFragment()).toMatchSnapshot()
})

test('it renders cases', () => {
  render(<Dashboard confirmed='123' recovered='456' deaths='789' />)
  const confirmedText = screen.getByText('123')
  expect(confirmedText).toBeInTheDocument()

  const recoveredText = screen.getByText('456')
  expect(recoveredText).toBeInTheDocument()

  const deathsText = screen.getByText('789')
  expect(deathsText).toBeInTheDocument()
})
