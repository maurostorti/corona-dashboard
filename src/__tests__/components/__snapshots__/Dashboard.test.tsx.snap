// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly confirmed 1`] = `
<DocumentFragment>
  <div
    class="css-e53awj-MuiStack-root"
  >
    <div
      class="css-nen11g-MuiStack-root"
    >
      <h2
        class="Data-title"
      >
         Confimed Cases
      </h2>
      <p
        class="Data-content"
      >
         123 
      </p>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
    />
    <div
      class="css-nen11g-MuiStack-root"
    >
      <h2
        class="Data-title"
      >
         Recovered Cases
      </h2>
      <p
        class="Data-content"
      >
        456
      </p>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
    />
    <div
      class="css-nen11g-MuiStack-root"
    >
      <h2
        class="Data-title"
      >
         Deaths
      </h2>
      <p
        class="Data-content"
      >
        789
      </p>
    </div>
  </div>
</DocumentFragment>
`;
